src/Raven.Server/Integrations/PostgreSQL/RqlQuery.cs

```diff
@@ -175,90 +188,104 @@ public static bool TryParse(string queryText, int[] parametersDataTypes, Documen
+                        {
+                            // remove @metadata
+                            jsonResult.Modifications.Remove(Constants.Documents.Metadata.Key);
+                        }
 
-                        HandleSpecialColumnsIfNeeded(columnName, prop, prop.Value, ref row);
-                        
-                        jsonResult.Modifications.Remove(columnName);
-                    }
+                        foreach (var (columnName, pgColumn) in Columns)
+                        {
+                            var index = jsonResult.GetPropertyIndex(columnName);
+                            if (index == -1)
+                                continue;
 
+                            jsonResult.GetPropertyByIndex(index, ref prop);
 
-                    if (jsonResult.Modifications.Removals.Count != jsonResult.Count)
-                    {
-                        using (DocumentDatabase.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
-                        using (context.OpenReadTransaction())
+                            var value = GetValueByType(prop, prop.Value, pgColumn);
+
+                            row[pgColumn.ColumnIndex] = value;
+
+                            HandleSpecialColumnsIfNeeded(columnName, prop, prop.Value, ref row);
+                            
+                            jsonResult.Modifications.Remove(columnName);
+                        }
+
+
+                        if (jsonResult.Modifications.Removals.Count != jsonResult.Count)
                         {
-                            var modified = context.ReadObject(jsonResult, "renew");
-                            row[jsonIndex] = Encoding.UTF8.GetBytes(modified.ToString());
+                            using (DocumentDatabase.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                            using (context.OpenReadTransaction())
+                            {
+                                var modified = context.ReadObject(jsonResult, "renew");
+                                row[jsonIndex] = Encoding.UTF8.GetBytes(modified.ToString());
+                            }
                         }
-                    }
 
-                    await writer.WriteAsync(builder.DataRow(row[..Columns.Count]), token);
+                        await writer.WriteAsync(builder.DataRow(row[..Columns.Count]), token);
+                    }
                 }
+                finally
+                {
+                    ArrayPool<ReadOnlyMemory<byte>?>.Shared.Return(row);
+                }
+
+                await writer.WriteAsync(builder.CommandComplete($"SELECT {_result.Count}"), token);
             }
             finally
             {
-                ArrayPool<ReadOnlyMemory<byte>?>.Shared.Return(row);
+                _queryOperationContext?.Dispose();
+                _queryOperationContext = null;
```

Comment:
Aren't you missing `_result = null;` here?

Reply:
I think it's necessary in order to hit this condition:

https://github.com/ravendb/ravendb/pull/15261/files#diff-de31fa030e49b5d9a259169d4bf9aef834714a07e75b4bb76e66397e5be3ad20R199-R202

Reply:
This also means that you could introduce a method like:

```
public voud ReleaseQueryResources()
{
           _queryOperationContext?.Dispose();
            _queryOperationContext = null;
            _result = null;
}
```

and use it here and in `Dispose()`

Reply:
done